CI: Builds failing in ``pandas.show_versions()``

Failing on all platofrms. E.g. 
https://github.com/pandas-dev/pandas/runs/4581277621

```
Run python -c "import pandas; pandas.show_versions();"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/pandas/pandas/pandas/util/_print_versions.py", line 109, in show_versions
    deps = _get_dependency_info()
  File "/home/runner/work/pandas/pandas/pandas/util/_print_versions.py", line 88, in _get_dependency_info
    mod = import_optional_dependency(modname, errors="ignore")
  File "/home/runner/work/pandas/pandas/pandas/compat/_optional.py", line 115, in import_optional_dependency
    module = importlib.import_module(name)
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1002, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 945, in _find_spec
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 83, in find_spec
    return method()
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 104, in spec_for_pip
    if self.pip_imported_during_build():
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 115, in pip_imported_during_build
    return any(
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 116, in <genexpr>
    frame.f_globals['__file__'].endswith('setup.py')
KeyError: '__file__'
Error: Process completed with exit code 1.
```

60.0.3 did not fix all problems for us unfortunately. When running this locally I get
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/testenv2/lib/python3.10/site-packages/pandas/util/_print_versions.py", line 109, in show_versions
    deps = _get_dependency_info()
  File "/tmp/testenv2/lib/python3.10/site-packages/pandas/util/_print_versions.py", line 88, in _get_dependency_info
    mod = import_optional_dependency(modname, errors="ignore")
  File "/tmp/testenv2/lib/python3.10/site-packages/pandas/compat/_optional.py", line 115, in import_optional_dependency
    module = importlib.import_module(name)
  File "/tmp/testenv2/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/tmp/testenv2/lib/python3.10/site-packages/setuptools/__init__.py", line 8, in <module>
    import _distutils_hack.override  # noqa: F401
  File "/tmp/testenv2/lib/python3.10/site-packages/_distutils_hack/override.py", line 1, in <module>
    __import__('_distutils_hack').do_override()
  File "/tmp/testenv2/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 76, in do_override
    ensure_local_distutils()
  File "/tmp/testenv2/lib/python3.10/site-packages/_distutils_hack/__init__.py", line 63, in ensure_local_distutils
    assert '_distutils' in core.__file__, core.__file__
AssertionError: /tmp/testenv2/lib/python3.10/distutils/core.py
```

It looks like it tries to distutils from stdlib, but if running

```
import importlib
importlib.import_module("setuptools")
```

it is pulling the distutils from 
```
'/tmp/testenv2/lib/python3.10/site-packages/setuptools/_distutils'
```

Not sure what is different when doing this via pandas.show_versions()

> This happens because show_version() imports all the dependencies installed including pip and setuptools. Apparently importing pip before setuptools causes this to happen. Swapping these too lines seems to fix it for me.
> 
> https://github.com/pandas-dev/pandas/blob/e18921eb0cc86f71c84a4aa0bd6d0c1b7de89def/pandas/util/_print_versions.py#L63-L64
> 
> 

This appears to be correct

```
import importlib

importlib.import_module("pip")
importlib.import_module("setuptools")
```
gives
```
...
File ~/miniconda3/envs/pandas-1.4.2/lib/python3.10/site-packages/_distutils_hack/__init__.py:59, in ensure_local_distutils()
     57 # check that submodules load as expected
     58 core = importlib.import_module('distutils.core')
---> 59 assert '_distutils' in core.__file__, core.__file__
     60 assert 'setuptools._distutils.log' not in sys.modules

AssertionError: /home/simon/miniconda3/envs/pandas-1.4.2/lib/python3.10/distutils/core.py
```

as now appearing in pre-release checks https://github.com/simonjayhawkins/pandas-release/runs/6552535479?check_suite_focus=true

rerunning the above code sample in a jupyter notebook, gives a warning second time round

```
/home/simon/miniconda3/envs/pandas-1.4.2/lib/python3.10/site-packages/_distutils_hack/__init__.py:17: UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
  warnings.warn(
/home/simon/miniconda3/envs/pandas-1.4.2/lib/python3.10/site-packages/_distutils_hack/__init__.py:30: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
```

> Perhaps its better to wait for an upstream fix, though?

switching order of imports, works fine...

```
import importlib

importlib.import_module("setuptools")
importlib.import_module("pip")
```

--- a/.github/workflows/python-dev.yml
+++ b/.github/workflows/python-dev.yml
@@ -54,11 +54,10 @@ jobs:
       with:
         python-version: '3.11-dev'
 
-    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
     - name: Install dependencies
       shell: bash -el {0}
       run: |
-        python -m pip install --upgrade pip "setuptools<60.0.0" wheel
+        python -m pip install --upgrade pip setuptools wheel
         pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
         pip install git+https://github.com/nedbat/coveragepy.git
         pip install cython python-dateutil pytz hypothesis pytest>=6.2.5 pytest-xdist pytest-cov

--- a/.github/workflows/sdist.yml
+++ b/.github/workflows/sdist.yml
@@ -41,10 +41,9 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip "setuptools<60.0.0" wheel
+        python -m pip install --upgrade pip setuptools wheel
 
         # GH 39416
         pip install numpy
@@ -66,10 +65,8 @@ jobs:
         channels: conda-forge
         python-version: '${{ matrix.python-version }}'
 
-    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
     - name: Install pandas from sdist
       run: |
-        python -m pip install --upgrade "setuptools<60.0.0"
         pip list
         python -m pip install dist/*.gz
 

--- a/ci/setup_env.sh
+++ b/ci/setup_env.sh
@@ -51,8 +51,7 @@ echo
 echo "update conda"
 conda config --set ssl_verify false
 conda config --set quiet true --set always_yes true --set changeps1 false
-# TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
-conda install -y -c conda-forge -n base 'mamba>=0.21.2' pip
+conda install -y -c conda-forge -n base 'mamba>=0.21.2' pip setuptools
 
 echo "conda info -a"
 conda info -a
@@ -67,8 +66,6 @@ echo "mamba env update --file=${ENV_FILE}"
 # See https://github.com/mamba-org/mamba/issues/633
 mamba create -q -n pandas-dev
 time mamba env update -n pandas-dev --file="${ENV_FILE}"
-# TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
-mamba install -n pandas-dev 'setuptools<60'
 
 echo "conda list -n pandas-dev"
 conda list -n pandas-dev

--- a/pandas/tests/util/test_show_versions.py
+++ b/pandas/tests/util/test_show_versions.py
@@ -7,7 +7,6 @@ import pytest
 from pandas.compat import (
     IS64,
     is_ci_environment,
-    is_numpy_dev,
 )
 from pandas.util._print_versions import (
     _get_dependency_info,
@@ -15,14 +14,6 @@ from pandas.util._print_versions import (
 )
 
 import pandas as pd
-
-# This is failing on the Numpy Dev build,
-# but the error may just be from distutils?
-pytestmark = pytest.mark.xfail(
-    is_numpy_dev,
-    reason="_distutils not in python3.10/distutils/core.py",
-    raises=AssertionError,
-)
 
 
 @pytest.mark.filterwarnings(

--- a/pandas/util/_print_versions.py
+++ b/pandas/util/_print_versions.py
@@ -60,8 +60,8 @@ def _get_dependency_info() -> dict[str, JSONSerializable]:
         "pytz",
         "dateutil",
         # install / build,
-        "pip",
         "setuptools",
+        "pip",
         "Cython",
         # test
         "pytest",

